### DISCUSSION TITLE
Feature request

### DISCUSSION THREAD

**Discussion by hamzaelmaghari** [laravel/framework#44051] [Q&A]
# Give the ability to customize for performance.

**Comment by Sophist-UK** [comment]
1. You need to be a lot clearer about what you are expecting such a feature to do to allow customization for performance.
    There are a lot of programming choices that lead to good performance (i.e. indexes, eager loading) and equally a lot of programming choices that lead to bad performance (e.g. lazy loading, n+1, filtering models rather than in SQL etc.), but this is about how you programme not about a feature to allow you to improve performance.
2. It would be helpful once you clarify top change the title of this discussion topic because "Feature request" is not very helpful. At a minimum "[Feature request] Performance customization" would be better, but ideally it would include a brief summary of the detail resulting from point 1.

**Reply by hamzaelmaghari** [reply]
You are right about what you've said here, the idea I got yesterday is when someone works with Vue or React. This means I don't need all those features that Laravel provides.

I mean what if Taylor make a feature on the installer which lets you specify which stack is going to be used for that project, then you get only what you need.

**Reply by henzeb** [reply]
There is already a laravel zero project, which is useful if you are going to build a console application. What stacks are you talking about specifically?

**Reply by Sophist-UK** [reply]
Laravel is already a selection of components. You can (to some extent) select what you want.

Laravel does NOT as far as I know come with Vue or React or even Livewire. You need to add these yourself.

All that said, I have in my head a Laravel Application Generator that will allow you to build an entire Laravel application to best programming standards entirely through a browser GUI. You will define tables, columns, relationships and indexes, and this tool will generate Models and Migrations and Seeders and Factories. You will define views with static text and lists / forms and the tool will write controllers and validation resources and create the matching Vue or React or Svelte or LiveWire or Tailwind code. Whether this will ever make it into useable code is another question.

**Reply by hamzaelmaghari** [reply]
> There is already a laravel zero project, which is useful if you are going to build a console application. What stacks are you talking about specifically?

APIs for example => vue & react

**Reply by hamzaelmaghari** [reply]
> Laravel is already a selection of components. You can (to some extent) select what you want.
> 
> Laravel does NOT as far as I know come with Vue or React or even Livewire. You need to add these yourself.
> 
> All that said, I have in my head a Laravel Application Generator that will allow you to build an entire Laravel application to best programming standards entirely through a browser GUI. You will define tables, columns, relationships and indexes, and this tool will generate Models and Migrations and Seeders and Factories. You will define views with static text and lists / forms and the tool will write controllers and validation resources and create the matching Vue or React or Svelte or LiveWire or Tailwind code. Whether this will ever make it into useable code is another question.

Not that far. just imagine I work with next js and I need Laravel only as an API ... then what are the importance of blade here!?

**Comment by henzeb** [comment]
There are a lot of options in laravel already. https://kinsta.com/blog/laravel-performance/
